returnn/tf/layers/basic.py

```diff
@@ -913,37 +913,199 @@ def transform_config_dict(cls, d, network, get_layer):
 
 class GatherLayer(_ConcatInputLayer):
   """
-  This is basically ``x[position]`` in some axis.
-  See also :class:`SliceLayer`.
-  See also :class:`GatherNdLayer`, which works on the time-axis, but supports different positions per batch.
+  Gathers slices on a specified axis from the input layer using indices from a ``position`` layer.
+  If the input is a layer of the shape ``[B,D,F1]``, and indices of shape ``[B,F2]``, this will yield output of the
+  shape ``[B,F2,F1]`` where
 
-  (Currently `position` can just be an integer, so it's much less generic than ``tf.gather``.
-   However, we might extend this to lists, arrays, or even other layers.)
+  ``output[b,f2,f1] = input[b,indices[b,f2],f1]``
+
+  (if ``D`` is the axis to gather from).
+  In general, all shared axes of the input and the positions will be considered as batch-axes.
+
+  It's basically a wrapper around ``tf.gather``.
+  See also :class:`GatherNdLayer`.
   """
   layer_class = "gather"
 
-  def __init__(self, axis, position, **kwargs):
+  def __init__(self, position, axis, **kwargs):
     """
-    :param str axis:
-    :param int position:
+    :param LayerBase position: Layer containing the indices used to select the slices of the input from.
```

Comment:
```suggestion
    :param int|LayerBase position: Layer containing the indices used to select the slices of the input from.
```

Reply:
is this how it should be? `__init__` itself can only handle `position` arguments of type `LayerBase`, not `int`. But yes, one can of course write it as a `int` within a config.

Reply:
`__init__` should be able to handle `int` directly.